examples/observe_qt.py

```diff
@@ -56,18 +98,43 @@ def __init__(self, state, *args, **kwargs):
         self.reset.clicked.connect(self.on_reset_clicked)
 
     def on_button_clicked(self):
-        self.state["clicked"] += 1
+        self.thread = QThread()
+        self.worker = LongJob()
+        self.worker.moveToThread(self.thread)
+        self.thread.started.connect(self.worker.run)
+        self.worker.finished.connect(self.thread.quit)
+        self.worker.finished.connect(self.worker.deleteLater)
+        self.thread.finished.connect(self.thread.deleteLater)
+
+        self.thread.start()
+
+        def progress(x):
+            self.state["progress"] = x
+
+        def bump(x):
+            self.state["clicked"] += x
+
+        self.button.setEnabled(False)
+        self.thread.finished.connect(lambda: self.button.setEnabled(True))
+        self.worker.result.connect(lambda x: bump(x))
+        self.worker.progress.connect(lambda x: progress(x))
 
     def on_reset_clicked(self):
         self.state["clicked"] = 0
 
 
 if __name__ == "__main__":
     # Define some state
-    state = observe({"clicked": 0})
+    state = observe({"clicked": 0, "progress": 0})
 
     app = QApplication([])
 
+    # Use a timer to run the scheduler
+    timer = QTimer()
+    timer.timeout.connect(scheduler.flush)
+    timer.setInterval(1000 / 60)
+    timer.start()
```

Comment:
In order to avoid event queue build up (when flush takes longer to run than 1000/60) we should instead schedule a flush only when necessary (on queueing), and not reschedule while it is still pending